### PR TITLE
Fix some phpstan 9 errors

### PR DESCRIPTION
**Pull request by VincentLanglet**
## Subject

Fixing almost 100 phpstan errors level 9.

During this, I discovered some errors about the type-alias we defined in Pool.
`item_adds` key is not an array of item, but just an array of string, according to the conf, the doc and the tests.

But in the AdminCompilerPass, we merge this array with the `items` array.
https://github.com/sonata-project/SonataAdminBundle/blob/2246f89b1c92750e8225f51544f261ab3d65de90/src/DependencyInjection/Compiler/AddDependencyCallsCompilerPass.php#L219

So we end with something like
image

But then, this is passed to the Pool, which consider the `items` key as full of arrays (which are type aliased as Item[]).
- array_map is called on `group['items']` in Pool
- items are considered as arrays in GroupMenuProvider

Does someone use the `item_adds` key @sonata-project/contributors ?
To me it cannot work. And I don't understand the purpose since we can use `item`. Should we deprecate the key ?

## Changelog

```markdown
### Deprecated
- Passing '' for the `admin` key in an item of the `sonata_admin.dashboard.groups` config
- Passing a `route` or a `label` when an `admin` is passed in an item of the `sonata_admin.dashboard.groups` config
- The `item_adds` key when configuring the `sonata_admin.dashboard.groups` config
```

### CODE REVIEW

**Comment by VincentLanglet** [comment]
When it was introduced
https://github.com/sonata-project/SonataAdminBundle/commit/f8616a82b4f2399d564f08eef2ceb6369853e010#diff-a4650f72c19f5835a8dea0883feacdc0d2bf22dc9cebb0d94e9b00aa75c6a647
the `item` node was a scalar too.

But it isn't now. So,
- either the `item_adds` config must be modified to be like the `item` one
- either we deprecate/remove the feature, since it's not working and nobody complained about it.

**Comment by jordisala1991** [comment]
Maybe the purpose is to add items to an already defined group by some other bundle? Not sure, I have never use it either.

To me we could deprecate it if it simplifies things. Or if it is completely broken, we might fix it (by removing it?),

File: src/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -24,6 +24,148 @@
...
+ *     pager_links: int|null,
+ *     skin: 'skin-black'|'skin-black-light'|'skin-blue'|'skin-blue-light'|'skin-green'|'skin-green-light'|'skin-purple'|'skin-purple-light'|'skin-red'|'skin-red-light'|'skin-yellow'|'skin-yellow-light',
+ *     sort_admins: bool,
+ *     use_bootlint: bool,
+ *     use_icheck: bool,
+ *     use_select2: bool,
+ *     use_stickyforms: bool,
+ * }
+ *
+ * @phpstan-type SonataAdminConfiguration = array{
```
Comment (core23) [inline src/DependencyInjection/Configuration.php:55]: This structure looks very complex. Maybe it's better to create a phpstan extension that can guess this structure based on the `Configuration` class.

**Comment by VincentLanglet** [comment]
> Maybe the purpose is to add items to an already defined group by some other bundle?

I understand it like this too.

> To me we could deprecate it if it simplifies things. Or if it is completely broken, we might fix it (by removing it?),

It's completely broken to me since we mixed up string and array. So I think I will remove it since nobody created an issue about it. And the day someone will request the feature, we can implement it correctly.

File: src/DependencyInjection/Configuration.php (hunk 1 above)
Comment (VincentLanglet) [reply src/DependencyInjection/Configuration.php:55]: To me, phpstan extensions are made to understand code, not generating some phpdoc.
And I need this type to have a correct phpdoc in tests or in some other classes.

Comment (jordisala1991) [reply src/DependencyInjection/Configuration.php:55]: To me it only reflects the complexity of the Sonata Admin Bundle configuration options. What should be correctly specified is the opcional keys. All this is always generated? I mean, if you use Sonata with the minimal configuration needed, still all the array is generated or some of it is omited?

Comment (VincentLanglet) [reply src/DependencyInjection/Configuration.php:55]: > All this is always generated? I mean, if you use Sonata with the minimal configuration needed, still all the array is generated or some of it is omited?

I need to check again but this is what I tried to defined, in order to write
```
/** @phpstan-var SonataAdminConfiguration $config */
$config = (new Processor())->processConfiguration(new Configuration(), []);
```

**Comment by VincentLanglet** [comment]
By looking at the `items` config, and dumping a basic config in test, I discovered the item type was wrong.
```
array{
 *     admin?: string,
 *     label: string,
 *     roles: list<string>,
 *     route: string,
 *     route_absolute: bool,
 *     route_params: array<string, string>
 * }
```
When you have an admin, you may not have a label/route key.

When using 
```
- admin: 'name'
```
you have to pass a route/label which is not used but 
```
- 'name'
```
will work as the previous code and won't have a route/label

So I changed the type to
```
array{
 *     label: string,
 *     roles: list<string>,
 *     route: string,
 *     route_absolute: bool,
 *     route_params: array<string, string>
 * }|{
 *     admin: string,
 *     roles: list<string>,
 *     route_absolute: bool,
 *     route_params: array<string, string>
 * }
```
And I deprecated
- Passing route/label when an admin is passed
- Passing `''` as an admin

**Comment by VincentLanglet** [comment]
I think this is ready to be reviewed/merged @jordisala1991

File: src/Request/AdminFetcher.php [hunk 2]
```diff
@@ -34,7 +34,7 @@ public function get(Request $request): AdminInterface
         $adminCode = $request->get('_sonata_admin');
 
         if (!\is_string($adminCode)) {
-            $route = $request->get('_route');
+            $route = $request->get('_route', '');
```
Comment (core23) [inline src/Request/AdminFetcher.php:37]:
```suggestion
            $route = $request->attributes->get('_route', '');
```

Comment (VincentLanglet) [reply src/Request/AdminFetcher.php:37]: Isn't this a BC break ? If someone provide a `_route` as a query param or so.
I thought we could only change those code on the next major to solve https://github.com/sonata-project/SonataAdminBundle/issues/7365

**Review by core23 — APPROVED** [review]

**Review by jordisala1991 — APPROVED** [review]

**Comment by jordisala1991** [comment]
Thanks!

**Pull request merged**